src/VisualStudio/Core/Def/Implementation/AbstractEditorFactory.cs

```diff
@@ -207,6 +251,56 @@ int IVsEditorFactory.Close()
             return LoaderName;
         }
 
+        private (string, Guid, IntPtr) GetWinFormsDesignerData(IVsHierarchy vsHierarchy, uint itemid, IVsTextBuffer? textBuffer, READONLYSTATUS readOnlyStatus, bool initializeView)
+        {
+            var loaderName = GetWinFormsLoaderName(vsHierarchy);
+            var designerService = (IVSMDDesignerService)_oleServiceProvider.QueryService<SVSMDDesignerService>();
+            var designerLoader = (IVSMDDesignerLoader)designerService.CreateDesignerLoader(loaderName);
+            if (designerLoader is null)
+            {
+                return GetCodeEditorData(textBuffer, readOnlyStatus, initializeView);
+            }
+
+            try
+            {
+                if (initializeView)
+                {
+                    designerLoader.Initialize(_oleServiceProvider, vsHierarchy, (int)itemid, textBuffer as IVsTextLines);
+                }
+                var editorCaption = designerLoader.GetEditorCaption((int)readOnlyStatus);
+
+                var designer = designerService.CreateDesigner(_oleServiceProvider, designerLoader);
+
+                var documentView = initializeView ? Marshal.GetIUnknownForObject(designer.View) : IntPtr.Zero;
+                var commandUI = designer.CommandGuid;
+
+                return (editorCaption, commandUI, documentView);
+            }
+            catch
+            {
+                designerLoader.Dispose();
+                throw;
+            }
+        }
+
+        private (string, Guid, IntPtr) GetCodeEditorData(IVsTextBuffer? textBuffer, READONLYSTATUS readOnlyStatus, bool initializeView)
```

Comment:
Named tuples please, as otherwise it's not obvious what these are.

Reply:
Done.